eng/tools/typespec-validation/src/rules/sdk-tspconfig-validation.ts

```diff
@@ -406,9 +429,47 @@ export class TspConfigCsharpAzPackageDirectorySubRule extends TspconfigEmitterOp
   }
 }
 
-export class TspConfigCsharpAzNamespaceEqualStringSubRule extends TspconfigEmitterOptionsSubRuleBase {
+export class TspConfigCsharpAzNamespaceEqualStringSubRule extends TspconfigSubRuleBase {
+  protected emitterName: string;
   constructor() {
-    super("@azure-tools/typespec-csharp", "namespace", "{package-dir}");
+    super("namespace", "{package-dir}");
+    this.emitterName = "@azure-tools/typespec-csharp";
+  }
+  protected validate(config: any): RuleResult {
+    let option: Record<string, any> | undefined = config?.options?.[this.emitterName];
+    for (const segment of this.keyToValidate.split(".")) {
+      if (option && typeof option === "object" && !Array.isArray(option) && segment in option)
+        option = option![segment];
+      else
+        return this.createFailedResult(
+          `Failed to find "options.${this.emitterName}.${this.keyToValidate}"`,
+          `Please add "options.${this.emitterName}.${this.keyToValidate}"`,
+        );
+    }
+
+    if (option === undefined)
+      return this.createFailedResult(
+        `Failed to find "options.${this.emitterName}.${this.keyToValidate}"`,
+        `Please add "options.${this.emitterName}.${this.keyToValidate}"`,
+      );
+
+    const packageDir = config?.options?.[this.emitterName]?.["package-dir"];
+    const actualValue = option as unknown as undefined | string | boolean;
+    if (
+      this.validateValue(actualValue, this.expectedValue) ||
+      (packageDir !== undefined && this.validateValue(actualValue, packageDir))
+    ) {
+      return { success: true };
+    }
+
+    return this.createFailedResult(
+      `The value of options.${this.emitterName}.${this.keyToValidate} "${actualValue}" does not match "${this.expectedValue}" or "${packageDir}"`,
+      `Please update the value of "options.${this.emitterName}.${this.keyToValidate}" to match "${this.expectedValue}" or "${packageDir}"`,
+    );
+  }
+
+  public getPathOfKeyToValidate() {
```

Comment:
you can extend `TspconfigEmitterOptionsSubRuleBase ` and override `validate` method, then you dont need to copy this method.